protocol/communities/permission_checker.go

```diff
@@ -106,7 +106,7 @@ func (p *DefaultPermissionChecker) CheckPermissionToJoin(community *Community, a
 
 	accountsAndChainIDs := combineAddressesAndChainIDs(addresses, allChainIDs)
 
-	if len(becomeMemberPermissions) == 0 || len(permissionsToJoin) == 0 {
+	if len(permissionsToJoin) == 0 {
```

Comment:
**permissionsToJoin** holds all permissions (admin, member and tokenMaster).
It is sufficient to check the length of it only.
Also, in case there was a tokenMaster permission only, this function returned a default response.

Reply:
It will not work.
1. Create a community
2. Add admin permission
3. Try to join as a simple member, which does not fit the admin permission